job-dsls/jobs/DailyBuild_pipeline.groovy

```diff
@@ -0,0 +1,613 @@
+    description("This job: <br> - Runs the KIE Server integration tests on mutiple supported containers and JDKs <br> IMPORTANT: Created automatically by Jenkins job DSL plugin. Do not edit manually! The changes will get lost next time the job is generated. ")
+
+    // Label which specifies which nodes this job can run on.
+    label("master")
+
+    parameters {
+        stringParam("kieVersion", "${kieVersion}", "please edit the version of the KIE release <br> i.e. typically <b> major.minor.micro.<extension> </b>7.1.0.Beta1 for <b> community </b>or <b> major.minor.micro.<yyymmdd>-productized </b>(7.1.0.20170514-productized) for <b> productization </b> <br> Version to test. Will be supplied by the parent job. <br> Normally the KIE_VERSION will be supplied by parent job <br> ******************************************************** <br> ")
+        stringParam("baseBranch", "${baseBranch}", "please edit the branch of the KIE release <br> Will be supplied by the parent job. <br> Normally the baseBranch will be supplied by parent job <br> ******************************************************** <br> ")
+
+    }
+
+    axes {
+        jdk("${javadk}")
+        text("container", "wildfly", "eap7", "tomcat9")
+        labelExpression("label_exp", "kie-linux&&kie-mem8g")
+    }
+
+    childCustomWorkspace("\${SHORT_COMBINATION}")
+
+    logRotator {
+        numToKeep(10)
+    }
+
+    wrappers {
+        timeout {
+            absolute(120)
+        }
+        timestamps()
+        colorizeOutput()
+        preBuildCleanup()
+        configFiles {
+            mavenSettings("771ff52a-a8b4-40e6-9b22-d54c7314aa1e") {
+                variable("SETTINGS_XML_FILE")
+                targetLocation("jenkins-settings.xml")
+            }
+        }
+    }
+
+    configure { project ->
+        project / 'buildWrappers' << 'org.jenkinsci.plugins.proccleaner.PreBuildCleanup' {
+            cleaner(class: 'org.jenkinsci.plugins.proccleaner.PsCleaner') {
+                killerType 'org.jenkinsci.plugins.proccleaner.PsAllKiller'
+                killer(class: 'org.jenkinsci.plugins.proccleaner.PsAllKiller')
+                username 'jenkins'
+            }
+        }
+    }
+
+    publishers {
+        archiveJunit("**/target/*-reports/TEST-*.xml")
+        mailer('mbiarnes@redhat.com', false, false)
+        wsCleanup()
+    }
+
+    steps {
+        shell(kieServerTest)
+        maven{
+            mavenInstallation("${mvnVersion}")
+            goals("-B -e -fae -nsu clean verify -P\$container -Pjenkins-pr-builder")
+            rootPOM("kie-server-parent/kie-server-tests/pom.xml")
+            properties("kie.server.testing.kjars.build.settings.xml":"\$SETTINGS_XML_FILE")
+            properties("maven.test.failure.ignore": true)
+            properties("deployment.timeout.millis":"240000")
+            properties("container.startstop.timeout.millis":"240000")
+            properties("eap7.download.url":EAP7_DOWNLOAD_URL)
```

Comment:
I am really not sure here, but should there be a $ sign before variable name?
```suggestion
            properties("eap7.download.url":"\$EAP7_DOWNLOAD_URL")
```

Reply:
I would put it here too  -but it is working like this in our daily builds (see https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/dailyBuild/job/kieServerMatrix-kieAllBuild-master/configure --> invike top level targets --> advanced and you will see here in properties:
eap7.download.url=http://download-ipv4.eng.brq.redhat.com/released/JBoss-middleware/eap7/7.2.0/jboss-eap-7.2.0.zip - so the variable is displayed right here

Reply:
OK, let's leave it as it is then.

Reply:
I see, it is a Groovy (JobDSL) variable, so it makes sense now.